tools/manifest/manifest.py

```diff
@@ -400,8 +400,18 @@ def load(tests_root, manifest, types=None, meta_filters=None):
     return _load(logger, tests_root, manifest, types, meta_filters)
 
 
+__load_cache = {}
+
+
 def _load(logger, tests_root, manifest, types=None, meta_filters=None):
     # "manifest" is a path or file-like object.
+    manifest_path = manifest if isinstance(manifest, string_types) \
+        else manifest.name
+    if manifest_path in __load_cache:
+        return __load_cache[manifest_path]
+    # Cache at most one manifest.
+    __load_cache.clear()
```

Comment:
Gecko uses two and servo uses three. I think we should allow one per url-base rather than one total.